src/server_manager/web_app/gcp_account.ts

```diff
@@ -83,9 +85,10 @@ export class GcpAccount implements gcp.Account {
     for (const response of listInstancesResponses) {
       const instances = response.items ?? [];
       instances.forEach((instance) => {
+        const {zoneId} = gcp_api.parseZoneUrl(instance.zone);
+        const locator = {projectId, zoneId, instanceId: instance.id};
         const id = `${this.id}:${instance.id}`;
-        const server = new GcpServer(id, instance, this.apiClient);
-        result.push(server);
+        result.push(new GcpServer(id, locator, instance.name, Promise.resolve(), this.apiClient));
```

Comment:
If the static IP has not been created, I think we should reject the creation promise, since the creation has failed.